Tests/Shared.Specs/StringAssertionSpecs.cs

```diff
@@ -1638,640 +1680,1747 @@ public void When_the_actual_string_equivalent_to_the_expected_but_with_trailing_
+                .And.ParamName.Should().Be("expected");
+        }
+
+        [Fact]
+        public void When_containment_at_most_once_is_asserted_against_an_empty_string_it_should_throw_earlier()
+        {
+            // Arrange
+            string actual = "a";
+            string expectedSubstring = "";
+
+            // Act
+            Action act = () => actual.Should().Contain(expectedSubstring, AtMost.Once());
+
+            // Assert
+            act.Should().Throw<ArgumentException>()
+                .WithMessage("Cannot assert string containment against an empty string.*")
+                .And.ParamName.Should().Be("expected");
+        }
+
+        [Fact]
+        public void When_string_containment_at_most_once_is_asserted_and_actual_value_is_null_then_it_should_throw()
+        {
+            // Arrange
+            string actual = null;
+            string expectedSubstring = "XYZ";
+
+            // Act
+            Action act = () => actual.Should().Contain(expectedSubstring, AtMost.Once());
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        #endregion
+
+        #region LessThan
+
+        [Fact]
+        public void When_string_containment_less_than_is_asserted_and_actual_value_contains_the_expected_string_less_than_expected_times_it_should_not_throw()
+        {
+            // Arrange
+            string actual = "ABCDEBCDF";
+            string expectedSubstring = "BCD";
+
+            // Act
+            Action act = () => actual.Should().Contain(expectedSubstring, LessThan.Times(3));
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_string_containment_less_than_is_asserted_and_actual_value_contains_the_expected_string_but_not_less_than_expected_times_it_should_throw()
+        {
+            // Arrange
+            string actual = "ABCDEBCDF";
+            string expectedSubstring = "BCD";
+
+            // Act
+            Action act = () => actual.Should().Contain(expectedSubstring, LessThan.Times(2), "that is {0}", "required");
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("Expected * \"ABCDEBCDF\" to contain \"BCD\" less than 2 times because that is required, but found 2 times.");
+        }
+
+        [Fact]
+        public void When_string_containment_less_than_twice_is_asserted_and_actual_value_does_not_contain_the_expected_string_it_should_not_throw()
+        {
+            // Arrange
+            string actual = "ABCDEF";
+            string expectedSubstring = "XYS";
+
+            // Act
+            Action act = () => actual.Should().Contain(expectedSubstring, LessThan.Twice());
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_containment_less_than_once_is_asserted_against_null_it_should_throw_earlier()
```

Comment:
The two tests of a null and empty `expected` are independent of the constraint, so there is no need to duplicate them for each of the constraints.
The same goes for testing the `because` and `becauseArgs`.